### PR TITLE
feat(log_artifact): add an option to copy (capture) into dvclive dir

### LINKED ISSUE #550: `log_artifact`: move model file (optionally or by default) into Live dir
In some (a lot?) of case models that are being produced are actually saved in some temporary locations, e.g. in case of YOLO it creates Git-ignored `runs/run15/weights/best.pt`. It doesn't make sense to keep model artifact there, it has to be moved somewhere. I think that we should have an option to move a model into `dvclive/models`.

I can make a PR for this. Just want to run by you @daavoo @dberenbaum @aguschin and see what you think.

### PR DESCRIPTION
**Pull request by shcheklein**
Closes #550

```python
    with Live() as live:
        live.log_artifact("model.pth", copy=True)
```

It copies a file into `dvclive/artifacts/model.pth` and tracks it with DVC in a regular way.

---------

- [X] ❗ I have followed the [Contributing to DVCLive](https://github.com/iterative/dvclive/blob/master/CONTRIBUTING.md) guide.

TODO (when accepted):

- [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I  have created a separate PR (or issue, at least) in
  [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: src/dvclive/live.py [hunk 1]
```diff
@@ -388,8 +393,15 @@ def log_artifact(
         if self._dvc_repo is not None:
             from dvc.repo.artifacts import name_is_compatible
 
+            if copy:
+                basename = os.path.basename(os.path.normpath(path))
+                output_dir = Path(self.artifacts_dir)
+                output_dir.mkdir(exist_ok=True, parents=True)
+                shutil.copy2(path, output_dir / basename)
```
Comment (daavoo) [inline src/dvclive/live.py:400]: We should also support directories here or, if we don't want to, log warning and not fail

**Review by daavoo — CHANGES_REQUESTED** [review]

File: src/dvclive/live.py [hunk 2]
```diff
@@ -380,6 +384,7 @@ def log_artifact(
         desc: Optional[str] = None,  # noqa: ARG002
         labels: Optional[List[str]] = None,  # noqa: ARG002
         meta: Optional[Dict[str, Any]] = None,  # noqa: ARG002
+        copy: bool = False,
```
Comment (dberenbaum) [inline src/dvclive/live.py:387]: What's the path that's being copied in your scenario, and what would happen if you logged it with `copy=False`? Could we infer whether to copy?

Comment (shcheklein) [reply src/dvclive/live.py:387]: > What's the path that's being copied in your scenario

The path that you pass to the logger: `log_artifact('some_path')`

> and what would happen if you logged it with copy=False?

Same behavior as before. `.dvc` created nearby that `some_path` and in `dvc.yaml` we log it as `some_path`.

> Could we infer whether to copy?

not sure tbh, sounds a bit too much magic / complicated? Probably we would need a bit more data to decide.

File: src/dvclive/live.py (hunk 1 above)
Comment (shcheklein) [reply src/dvclive/live.py:400]: yep, my bad - I though it's supported by `copy2`. Looking into this ...

File: src/dvclive/live.py (hunk 2 above)
Comment (dberenbaum) [reply src/dvclive/live.py:387]: > > What's the path that's being copied in your scenario
> 
> The path that you pass to the logger: `log_artifact('some_path')`

Sorry, I meant for the example you are working on that prompted this PR. What is the exact path you want to log?

I'm wondering if it's inside/outside the repo, gitignored, etc.

**Review by daavoo — APPROVED** [review]

**Pull request merged**